apps/game_backend/lib/game_backend/configuration.ex

```diff
@@ -320,13 +298,25 @@ defmodule GameBackend.Configuration do
 
   ## Examples
 
-      iex> get_latest_version()
+      iex> get_current_version()
       %Version{}
   """
 
   def get_current_version do
     q = from(v in Version, where: v.current)
+
     Repo.one(q)
+    |> Repo.preload(
+      characters: [
+        basic_skill: [mechanics: [:on_arrival_mechanic, :on_explode_mechanics, :parent_mechanic]],
+        ultimate_skill: [mechanics: [:on_arrival_mechanic, :on_explode_mechanics, :parent_mechanic]],
+        dash_skill: [mechanics: [:on_arrival_mechanic, :on_explode_mechanics, :parent_mechanic]]
+      ]
+    )
+    |> Repo.preload(:consumable_items)
+    |> Repo.preload(:skills)
+    |> Repo.preload(:map_configurations)
+    |> Repo.preload(:game_configuration)
```

Comment:
Isn't this doing 5 queries instead of 1? why not adding a preload list to the q variable?

Reply:
Doesn't matter too much adding it to the base query as preloads
```elixir
q = from v in Version, preload: [:consumable_items, :characters]
Repo.all(q)

[debug] QUERY OK source="versions" db=7.8ms queue=0.1ms idle=1010.8ms
SELECT v0."id", v0."name", v0."current", v0."inserted_at", v0."updated_at" FROM "versions" AS v0 []
↳ :elixir.eval_external_handler/3, at: src/elixir.erl:405
[debug] QUERY OK source="consumable_items" db=2.4ms idle=1019.2ms
SELECT c0."id", c0."active", c0."name", c0."radius", c0."effects", c0."mechanics", c0."version_id", c0."inserted_at", c0."updated_at", c0."version_id" FROM "consumable_items" AS c0 WHERE (c0."version_id" = $1) ORDER BY c0."version_id" ["85601af6-b6f2-4277-bd4f-f616445a03c0"]
↳ :elixir.eval_external_handler/3, at: src/elixir.erl:405
[debug] QUERY OK source="characters" db=8.7ms idle=1019.2ms
SELECT c0."id", c0."game_id", c0."active", c0."name", c0."title", c0."lore", c0."faction", c0."class", c0."quality", c0."ranks_dropped_in", c0."base_health", c0."base_attack", c0."base_defense", c0."base_stamina", c0."stamina_interval", c0."max_inventory_size", c0."natural_healing_interval", c0."natural_healing_damage_interval", c0."base_speed", c0."base_size", c0."basic_skill_id", c0."ultimate_skill_id", c0."dash_skill_id", c0."version_id", c0."inserted_at", c0."updated_at", c0."version_id" FROM "characters" AS c0 WHERE (c0."version_id" = $1) ORDER BY c0."version_id" ["85601af6-b6f2-4277-bd4f-f616445a03c0"]
↳ :elixir.eval_external_handler/3, at: src/elixir.erl:405
```

VS
```elixir
Repo.all(Version) |> Repo.preload([:consumable_items, :characters])

[debug] QUERY OK source="versions" db=3.4ms idle=1062.4ms
SELECT v0."id", v0."name", v0."current", v0."inserted_at", v0."updated_at" FROM "versions" AS v0 []
↳ :elixir.eval_external_handler/3, at: src/elixir.erl:405
[debug] QUERY OK source="consumable_items" db=8.6ms idle=1066.3ms
SELECT c0."id", c0."active", c0."name", c0."radius", c0."effects", c0."mechanics", c0."version_id", c0."inserted_at", c0."updated_at", c0."version_id" FROM "consumable_items" AS c0 WHERE (c0."version_id" = $1) ORDER BY c0."version_id" ["85601af6-b6f2-4277-bd4f-f616445a03c0"]
↳ :elixir.eval_external_handler/3, at: src/elixir.erl:405
[debug] QUERY OK source="characters" db=8.8ms idle=1066.3ms
SELECT c0."id", c0."game_id", c0."active", c0."name", c0."title", c0."lore", c0."faction", c0."class", c0."quality", c0."ranks_dropped_in", c0."base_health", c0."base_attack", c0."base_defense", c0."base_stamina", c0."stamina_interval", c0."max_inventory_size", c0."natural_healing_interval", c0."natural_healing_damage_interval", c0."base_speed", c0."base_size", c0."basic_skill_id", c0."ultimate_skill_id", c0."dash_skill_id", c0."version_id", c0."inserted_at", c0."updated_at", c0."version_id" FROM "characters" AS c0 WHERE (c0."version_id" = $1) ORDER BY c0."version_id" ["85601af6-b6f2-4277-bd4f-f616445a03c0"]
↳ :elixir.eval_external_handler/3, at: src/elixir.erl:405
```

Instead, if we add the join expression to the query this is what happens
```elixir
q = from v in Version, join: c in assoc(v, :characters), join: ci in assoc(v, :consumable_items), preload: [[characters: c], [consumable_items: ci]]
[debug] QUERY OK source="versions" db=6.8ms queue=5.7ms idle=1601.0ms
SELECT v0."id", v0."name", v0."current", v0."inserted_at", v0."updated_at", c1."id", c1."game_id", c1."active", c1."name", c1."title", c1."lore", c1."faction", c1."class", c1."quality", c1."ranks_dropped_in", c1."base_health", c1."base_attack", c1."base_defense", c1."base_stamina", c1."stamina_interval", c1."max_inventory_size", c1."natural_healing_interval", c1."natural_healing_damage_interval", c1."base_speed", c1."base_size", c1."basic_skill_id", c1."ultimate_skill_id", c1."dash_skill_id", c1."version_id", c1."inserted_at", c1."updated_at", c2."id", c2."active", c2."name", c2."radius", c2."effects", c2."mechanics", c2."version_id", c2."inserted_at", c2."updated_at" FROM "versions" AS v0 INNER JOIN "characters" AS c1 ON c1."version_id" = v0."id" INNER JOIN "consumable_items" AS c2 ON c2."version_id" = v0."id" []
```

Reply:
So, I think it is an optimization we don't need yet. But I can add the preloads to the base query as well

Reply:
Done!

Reply:
oh okey, i though that doing a `Repo.all/1` and the doing a `Repo.preload/2` was consulting the db twice when the preload just consults once